homeassistant/components/soma/cover.py

```diff
@@ -21,77 +38,179 @@ async def async_setup_entry(
+
+    async_add_entities(entities, True)
+
+
+class SomaTilt(SomaEntity, CoverEntity):
+    """Representation of a Soma Tilt device."""
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_BLIND
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = (
+            SUPPORT_OPEN_TILT
+            | SUPPORT_CLOSE_TILT
+            | SUPPORT_STOP_TILT
+            | SUPPORT_SET_TILT_POSITION
+        )
+        return supported_features
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return the current cover tilt position."""
+        return self.current_position
+
+    @property
+    def is_closed(self):
+        """Return if the cover tilt is closed."""
+        return self.current_position == 0
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], 100)
+        if is_api_response_success(response):
+            self.current_position = 0
+        else:
+            _LOGGER.error(MSG_DEVICE_UNREACHABLE, self.device["name"], response["msg"])
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], -100)
+        if is_api_response_success(response):
+            self.current_position = 100
+        else:
+            _LOGGER.error(MSG_DEVICE_UNREACHABLE, self.device["name"], response["msg"])
+
+    def stop_cover_tilt(self, **kwargs):
+        """Stop the cover tilt."""
+        response = self.api.stop_shade(self.device["mac"])
+        if is_api_response_success(response):
+            # Set cover position to some value where up/down are both enabled
+            self.current_position = 50
+        else:
+            _LOGGER.error(MSG_DEVICE_UNREACHABLE, self.device["name"], response["msg"])
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        # 0 -> Closed down (api: 100)
+        # 50 -> Fully open (api: 0)
+        # 100 -> Closed up (api: -100)
+        target_api_position = 100 - ((kwargs[ATTR_TILT_POSITION] / 50) * 100)
+        response = self.api.set_shade_position(self.device["mac"], target_api_position)
+        if is_api_response_success(response):
+            self.current_position = kwargs[ATTR_TILT_POSITION]
+        else:
+            _LOGGER.error(MSG_DEVICE_UNREACHABLE, self.device["name"], response["msg"])
+
+    async def async_update(self):
+        """Update the entity with the latest data."""
+        try:
+            _LOGGER.debug("Soma Tilt Update")
+            response = await self.hass.async_add_executor_job(
+                self.api.get_shade_state, self.device["mac"]
+            )
+        except RequestException:
+            _LOGGER.error(MSG_API_UNREACHABLE)
```

Comment:
https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=log&_highlight=once#silver-

Usually we only log once at `warning` level when it becomes unreachable, and once again at `info` when it reconnects

Reply:
Fixed in last commit, let me know what you think!